plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/IcebergRestCatalogConfig.java

```diff
@@ -81,4 +83,17 @@ public IcebergRestCatalogConfig setSessionType(SessionType sessionType)
         this.sessionType = sessionType;
         return this;
     }
+
+    public Optional<String> getWarehouse()
+    {
+        return warehouse;
+    }
+
+    @Config("iceberg.rest-catalog.warehouse")
+    @ConfigDescription("The warehouse location/identifier to use with the REST catalog server")
+    public IcebergRestCatalogConfig setWarehouse(String warehouse)
+    {
+        this.warehouse = Optional.ofNullable(warehouse);
```

Comment:
Let's sync this with `io.trino.plugin.iceberg.catalog.jdbc.IcebergJdbcCatalogConfig#setDefaultWarehouseDir`

(config name, description, property name, field name)

Reply:
Similar to the comment below, the REST catalog makes this a little more flexible and doesn't necessarily require an actual directory (it could just be a name that the catalog maps to a directory).